contracts/bridge_test.go

```diff
@@ -437,23 +450,59 @@ func TestExtendedBridgeAndCallback(t *testing.T) {
 	assert.NoError(t, err)
 	defer registerOfferEventSub.Unsubscribe()
 
-	// Subscribe bridge contract event
+	// Subscribe bridge contract events
 	b, err := bridge.NewBridge(bridgeAddr, backend) // create base bridge contract object, not extBridge object
 	assert.NoError(t, err)
+
+	requestValueTransferEventCh := make(chan *bridge.BridgeRequestValueTransfer, 10)
+	requestSub, err := b.WatchRequestValueTransfer(nil, requestValueTransferEventCh)
+	assert.NoError(t, err)
+	defer requestSub.Unsubscribe()
+
 	handleValueTransferEventCh := make(chan *bridge.BridgeHandleValueTransfer, 10)
 	handleSub, err := b.WatchHandleValueTransfer(nil, handleValueTransferEventCh)
 	assert.NoError(t, err)
 	defer handleSub.Unsubscribe()
 
-	// HandleERC20Transfer
-	amount := big.NewInt(1000)
+	// Approve / RequetERC20Transfer
 	rNonce := uint64(0)
-	rBlockNumber := uint64(0)
-	tx, err = b.HandleERC20Transfer(bridgeAccount, common.Hash{10}, aliceAcc.From, bobAcc.From, erc20Addr, amount, rNonce, rBlockNumber)
-	assert.Equal(t, nil, err)
+	amount := big.NewInt(1000)
+	offerPrice := big.NewInt(10)
+	dummyBigInt := big.NewInt(20)
+	extraData := []*big.Int{offerPrice, dummyBigInt}
+
+	tx, err = erc20.Approve(aliceAcc, bridgeAddr, amount)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, WaitMined(tx, backend, t))
+
+	tx, err = b.RequestERC20Transfer(aliceAcc, erc20Addr, bobAcc.From, amount, common.Big0, extraData)
+	assert.NoError(t, err)
 	backend.Commit()
 	assert.Nil(t, WaitMined(tx, backend, t))
 
+	// Check request request event
+	select {
+	case ev := <-requestValueTransferEventCh:
+		assert.Equal(t, amount.String(), ev.ValueOrTokenId.String())
+		assert.Equal(t, rNonce, ev.RequestNonce)
+		assert.Equal(t, erc20Addr, ev.TokenAddress)
+		assert.Equal(t, sc.ERC20, ev.TokenType)
+		assert.Equal(t, bobAcc.From, ev.To)
+		assert.Equal(t, len(extraData), len(ev.ExtraData))
+		assert.Equal(t, extraData[0].String(), ev.ExtraData[0].String())
+		assert.Equal(t, extraData[1].String(), ev.ExtraData[1].String())
+
+		// HandleERC20Transfer
+		tx, err = b.HandleERC20Transfer(bridgeAccount, ev.Raw.TxHash, ev.From, ev.To, ev.TokenAddress, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.ExtraData)
+		assert.Equal(t, nil, err)
```

Comment:
assert.NoError(t, err)
```suggestion
		assert.NoError(t, err)
```